stable/oauth2-proxy/Chart.yaml

```diff
@@ -1,5 +1,5 @@
 name: oauth2-proxy
-version: 1.1.0
+version: 1.1.1
```

Comment:
Do we consider this a bugfix? a backwards compatible feature or a non-backwards compatible change?

Reply:
This is a good question.  I originally thought that upgrading the apiVersion would be non-invasive, but [this issue](https://github.com/helm/helm/issues/6583) suggests that doing so it a breaking change, and there is no way around it.

Given that, it may be a non-backwards compatible change.  It needs to be done at some point though.  Right now, the chart will not install on kubernetes 1.16.

What do you think?  We can bump this chart to 1.2.0, or even 2.0.0?

Reply:
In doubt, it is better to bumb major version!

Reply:
Can `Capabilities` be used https://helm.sh/docs/developing_charts/ in order to detect version and choose the correct resource API based on that? That would make it smoother to transition?

Reply:
Hi @davidkarlsen, I considered this but thought hard-coding the `apps/v1` in the deployment was 'safe' because:

* `apps/v1` was introduced in kubernetes v1.9 in Dec 2017
* looking at the other stable helm charts revealed the following:
   * Out of the 332 charts that have a `Deployment` object, 231 are already hard-coded to use `apps/v1`
   * Out of the 332 charts that have a `Deployment` object, only **6** are using the `Capabilities` approach to set the apiVersion depending on the kubernetes version.

Given the above, it didn't seem like a good idea to introduce possibly more complexity into the chart needlessly.

If it's wanted, we can certainly do that for this chart, though.